### PR TITLE
[FSSS-153] Chore: Uses SliderOver component on Menu

### PR DESCRIPTION
**Pull request by hellofanny**
## What's the purpose of this pull request?

- [x] Uses SlideOver component on Menu
- [x] Visual adjustments on Menu
- [x] Avoids double scroll when using SlideOver
- [x] Attempts to hide first focus style (we're keeping it now, it's more a aesthetics "problem" over accessibility)

### **Menu**
- Uses SlideOver component on Menu (solves focus trap issue)
- Visual adjustments on Menu

|Before|After|
|-|-|
menu-before|menu-after

### **Double Scroll Issue**
|Before|After|
|-|-|
filters-before|filters-after

## How to test it?
You can test on the [preview link](https://preview-272--base.preview.vtex.app/):
On mobile:
- Open the menu.
- Open Filter and try to scroll.

### CODE REVIEW

File: src/components/common/Navbar/Navbar.tsx [hunk 1]
```diff
@@ -22,12 +25,38 @@ const links = [
...
 function Navbar() {
   const [showMenu, setShowMenu] = useState(false)
+  const [isMobile, setIsMobile] = useState<boolean>(false)
+  const { width: screenWidth } = useWindowDimensions()
+  let onDismissTransition: () => unknown
+
+  useEffect(() => {
+    if (screenWidth) {
+      // notebook breakpoint = 1280px (See breakpoints on styles/global.scss)
+      setIsMobile(screenWidth < 1280)
```
Comment (filipewl) [inline src/components/common/Navbar/Navbar.tsx:53]: Here's something that worked for me in another [PR](https://github.com/vtex-sites/base.store/blob/684147555bd45e985690f721aca8799a6fafbb4f/src/components/sections/ProductGallery/ProductGallery.tsx#L34-L39):

```suggestion
      const notebookBreakpoint =
	      getComputedStyle(document.documentElement).getPropertyValue(
	        '--breakpoint-notebook'
	      ) || '1280'
      setIsMobile(screenWidth < parseInt(notebookBreakpoint, 10))
```

**Review by filipewl — APPROVED** [review]
Looking good!

File: src/components/common/Navbar/Navbar.tsx (hunk 1 above)
Comment (ArthurTriis1) [reply src/components/common/Navbar/Navbar.tsx:53]: Isn't better to remove this logic to a custom hook? Something like this...

```
const useIsMobile = () => {
	 const [isMobile, setIsMobile] = useState<boolean>(false)

  useEffect(() => {
    if (!screenWidth) {
      return
    }

    const notebookBreakpoint =
      getComputedStyle(document.documentElement).getPropertyValue(
        '--breakpoint-notebook'
      ) || '1280'

    setIsMobile(screenWidth < parseInt(notebookBreakpoint, 10))
  }, [screenWidth])

return isMobile
}
```

**Review by ArthurTriis1 — APPROVED** [review]

File: src/components/common/Navbar/Navbar.tsx (hunk 1 above)
Comment (filipewl) [reply src/components/common/Navbar/Navbar.tsx:53]: I like this idea of the custom hook because I think we'll need it in more places.

Comment (hellofanny) [reply src/components/common/Navbar/Navbar.tsx:53]: Me too! Love it! I will try that! Thanks guysss 🥳

Comment (hellofanny) [reply src/components/common/Navbar/Navbar.tsx:53]: I decided to export a `isMobile` variable from `useWindowDimensions` hook :)  it'll be easier to apply wherever we need.

**Review by lucasfp13 — APPROVED** [review]
Great!! 💯

**Pull request merged**